### PR TITLE
3491 move plugins to earthly

### PR DESCRIPTION
**Pull request by aronchick**
## Summary by CodeRabbit

- **New Features**
    - Introduced Docker build stages for Python development in `Earthfile`.
    - Set up Travis CI for Python projects, specifying Python versions for testing.
    - Automated Git push process with `git_push.sh` script.
- **Bug Fixes**
    - Downgraded Swagger codegen version for compatibility.
- **Documentation**
    - Updated author information and Python version requirements in the Python client documentation.
    - Updated API and package version references in `README.md`.
- **Chores**
    - Updated `.gitignore` with new exclusions.
    - Enhanced Makefile configurations for better build and test processes.
    - Added sample configuration files for PyPI version and token.
- **Refactor**
    - Simplified build and clean targets in `Makefile`.
    - Updated version information handling in `setup.py`.

### CODE REVIEW

File: clients/python/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# ref: https://docs.travis-ci.com/user/languages/python
+language: python
+python:
+  - "3.2"
+  - "3.3"
+  - "3.4"
+  - "3.5"
```
[the remark this answers is not in the document]
Comment (aronchick) [reply clients/python/.travis.yml:7]: done

**Review by rossjones — APPROVED** [review]
Looks good, although the empty json files in pkg/repo/migrations/testdata/v2_custom_stores are tripping up the JSON linting

File: Makefile [hunk 2]
```diff
@@ -2,6 +2,14 @@ export GO = go
 export GOOS ?= $(shell $(GO) env GOOS)
 export GOARCH ?= $(shell $(GO) env GOARCH)
 
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Darwin)
+export SED = gsed
```
[the remark this answers is not in the document]
Comment (aronchick) [reply Makefile:7]: no - but the version of `sed` on macs is apparently old and causes bugs. I guess i'll build this into an error message.

File: clients/pyproject.toml.template [hunk 3]
```diff
@@ -6,11 +6,11 @@ build-backend = "hatchling.build"
 name = "bacalhau_apiclient"
 version = "VERSION"
 authors = [
-{ name="Enrico Rotundo", email="enrico.rotundo@gmail.com" },
+{ name="Bacalhau Team", email="support@bacalhau.org" },
```
[the remark this answers is not in the document]
Comment (aronchick) [reply clients/pyproject.toml.template:9]: Yeah, it comes to me

**Review by frrist — COMMENTED** [review]
Drive-by comment as I see the number of tools we are using to build/test bacalhau is increasing with this PR and I have some concerns. After this merges we will be using:
1. Github Actions for general orchestration of tests and linting
2. CircleCI for testing called via github actions
3. Earthly for building WebUI and Bacalhau plugins (latter being the main change here?)
4. TravisCI for testing python SDK(?)

I'd really like to see us narrow down the set of tools we are using. Perhaps we should do some re-thinking in follow ons to this?

**Pull request merged**